common/remote/rmtsmtp.cpp

```diff
@@ -633,6 +648,74 @@ class CMailInfo
         out.append("HELO ").append(mailServer.get()).append("\r\n");
     }
 
+    void getEhlo(StringBuffer & out) const
+    {
+        out.append("EHLO ").append(mailServer.get()).append("\r\n");
+    }
+
+    void getEhloAndOptions(StringBuffer & out)
+    {
+        // Initial EHLO and gather reply, which should contain ESMTP options
+        getEhlo(out);
+        writeAndAck(out.str(), out.length());
+
+#if defined(_USE_OPENSSL)
+        bool useTLS = false;
+        if (inlen > 0)
```

Comment:
if (inlen > 12), otherwise a very short response could access invalid memory

Reply:
The while condition a couple of lines down guards against an invalid access. However, this test can be modified slightly to ensure that a three-digit code and the required space|dash is available.